### PR TITLE
Cluster: Generate a new cluster certificate when bootstrapping a cluster

### PR DESCRIPTION
**Pull request by tomponline**
Rather than re-using the existing first-member server certificate.

### CODE REVIEW

File: lxd/cluster/membership.go [hunk 1]
```diff
@@ -116,18 +116,20 @@ func Bootstrap(state *state.State, gateway *Gateway, name string) error {
...
 
-	// The cluster certificates are symlinks against the regular node
-	// certificate.
-	for _, ext := range []string{".crt", ".key", ".ca"} {
-		if ext == ".ca" && !shared.PathExists(filepath.Join(state.OS.VarDir, "server.ca")) {
-			continue
-		}
-		err := os.Symlink("server"+ext, filepath.Join(state.OS.VarDir, "cluster"+ext))
+	// The cluster CA certificate is a symlink against the regular server CA certificate.
+	if shared.PathExists(filepath.Join(state.OS.VarDir, "server.ca")) {
```
Comment (tomponline) [inline lxd/cluster/membership.go:120]: @stgraber I wasn't sure about this bit. If there is a CA cert available, should the newly generated cluster certificate be signed by that CA cert?

Comment (stgraber) [reply lxd/cluster/membership.go:120]: No, we don't have a key for .ca, it's just a certificate used to check client connections.

**Comment by stgraber** [comment]
@tomponline looks like this got Jenkins stuck?

**Comment by tomponline** [comment]
@stgraber indeed, will run this locally and see if I can figure out whats going on here. Seemed like this would be a simple change :)

**Comment by tomponline** [comment]
@stgraber does this look like a bug to you?

https://github.com/lxc/lxd/blob/f12f03a4ba4645892ef6cc167c24da49d1217b02/lxd/endpoints/cluster.go#L48

If one calls `ClusterUpdateAddress()` with an address that is "covered" by the current listen address, it will close the listener and return no error? Seems odd.

**Comment by stgraber** [comment]
There are two addresses for clusters, core.https_address and cluster.https_address. If the latter is covered by the former, the listener must be turned off to avoid a conflict.

**Comment by tomponline** [comment]
@stgraber actually ignore me, I'm not familiar with the intricacies of "cluster" vs "network" addresses, digging deep into that now

**Comment by tomponline** [comment]
@stgraber thanks, yeah makes sense. I think the issue is that now we're generating a new cluster cert on bootstrap I need to restart the listeners and the so-called "gateway" (not sure what that is, I think its the actual dqlite socket?) so that they use the new cluster cert.

**Comment by tomponline** [comment]
Ah there's an `endpoints.NetworkUpdateCert()` perfect :)

**Comment by stgraber** [comment]
Failing CI, marking incomplete for now

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**